js/MediaStream.ts

```diff
@@ -46,50 +59,46 @@ function MediaStream(arg, id) {
 	if (
 		(!(arg instanceof window.Blob) &&
 			arg instanceof originalMediaStream &&
-			typeof arg.getBlobId === 'undefined') ||
+			!('getBlobId' in arg)) ||
 		(Array.isArray(arg) && arg[0] instanceof originalMediaStreamTrack)
 	) {
-		return new originalMediaStream(arg);
+		return new originalMediaStream(arg as any) as MediaStreamShim;
 	}
 
 	// new MediaStream(MediaStream) // stream
 	// new MediaStream(MediaStreamTrack[]) // tracks
 	// new MediaStream() // empty
 
 	id = id || newMediaStreamId();
-	var blobId = 'MediaStream_' + id;
+	const blobId = 'MediaStream_' + id;
 
 	// Extend returned MediaTream with custom MediaStream
-	var stream;
+	let stream: MediaStreamShim;
 	if (originalMediaStream !== window.Blob) {
-		stream = new (Function.prototype.bind.apply(originalMediaStream.bind(this), []))();
+		// Using native MediaStream class
+		stream = new (Function.prototype.bind.apply(originalMediaStream.bind(this), [[]]))();
 	} else {
 		// Fallback on Blob if originalMediaStream is not a MediaStream and Emulate EventTarget
-		stream = new Blob([blobId], {
+		stream = (new Blob([blobId], {
 			type: 'stream'
-		});
+		}) as any) as MediaStreamShim;
 
-		var target = document.createTextNode(null);
+		const target = document.createTextNode('');
 		stream.addEventListener = target.addEventListener.bind(target);
 		stream.removeEventListener = target.removeEventListener.bind(target);
 		stream.dispatchEvent = target.dispatchEvent.bind(target);
 	}
 
-	Object.defineProperties(stream, Object.getOwnPropertyDescriptors(MediaStream.prototype));
+	// Assign all members from the MediaStreamWrapper prototype to extend functionality of the Blob or MediaStream
+	Object.defineProperties(stream, Object.getOwnPropertyDescriptors(MediaStreamShim.prototype));
 
-	// Make it an EventTarget.
-	EventTarget.call(stream);
```

Comment:
Make sure this still works.

Reply:
ok hopefull TypeScript extends and super() take are of that.

Reply:
It works on all of them except for MediaStream.  Working on a fix.

Reply:
ah, you may add "EventTarget.call(stream);" again then.

Reply:
Fixed in 0f29fc0.  Not quite as simple as before because of the inheritance differences, but it works well.